tools/pd-ctl/pdctl/command/keyspace_command.go

```diff
@@ -15,33 +15,216 @@
 package command
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
 	"net/http"
 
 	"github.com/spf13/cobra"
+	"github.com/tikv/pd/server/apiv2/handlers"
 )
 
-const keyspacePrefix = "pd/api/v2/keyspaces"
+const (
+	keyspacePrefix = "pd/api/v2/keyspaces"
+	// flags
+	nmID        = "id"
+	nmName      = "name"
+	nmConfig    = "config"
+	nmLimit     = "limit"
+	nmPageToken = "page-token"
+)
 
 // NewKeyspaceCommand returns a keyspace subcommand of rootCmd.
 func NewKeyspaceCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "keyspace [command] [flags]",
-		Short: "show keyspace information",
-		Run:   showKeyspaceCommandFunc,
+		Use:   "keyspace <command> [flags]",
+		Short: "keyspace commands",
 	}
+	cmd.AddCommand(newShowKeyspaceCommand())
+	cmd.AddCommand(newCreateKeyspaceCommand())
+	cmd.AddCommand(newUpdateKeyspaceConfigCommand())
+	cmd.AddCommand(newUpdateKeyspaceStateCommand())
+	cmd.AddCommand(newListKeyspaceCommand())
 	return cmd
 }
 
+func newShowKeyspaceCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "show [flags]",
+		Short: "show keyspace metadata specified by keyspace name/id, if both provided, name will be used",
+		Run:   showKeyspaceCommandFunc,
+	}
+	r.Flags().String(nmID, "", "keyspace id")
+	r.Flags().String(nmName, "", "keyspace name")
```

Comment:
I prefer to use the subcommand here so that we can avoid setting them at the same time.

Reply:
done